Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp

```diff
@@ -863,4 +855,81 @@ void CanvasRenderingContext2D::paint_shadow_for_stroke_internal(Gfx::Path const&
     did_draw(path.bounding_box());
 }
 
+String CanvasRenderingContext2D::filter() const
+{
+    if (!drawing_state().filters_style_value) {
+        return String::from_utf8_without_validation("none"sv.bytes());
+    }
+
+    // Note: Should it be computed?
+    return drawing_state().filters_style_value->to_string(CSS::CSSStyleValue::SerializationMode::Normal);
+}
+
+// https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-filter
+void CanvasRenderingContext2D::set_filter(String filter)
+{
+    drawing_state().filters.clear();
+
+    // 1. If the given value is "none", then set this's current filter to "none" and return.
+    if (filter == "none"sv) {
+        return;
+    }
+
+    auto& realm = static_cast<CanvasRenderingContext2D&>(*this).realm();
+    auto parser = CSS::Parser::Parser::create(CSS::Parser::ParsingContext(realm), filter);
+
+    // 2. Let parsedValue be the result of parsing the given values as a <filter-value-list>.
+    //    If any property-independent style sheet syntax like 'inherit' or 'initial' is present,
+    //    then this parsing must return failure.
+    auto style_value = parser.parse_as_css_value(CSS::PropertyID::Filter);
+
+    if (style_value && style_value->is_filter_value_list()) {
+        auto filter_value_list = style_value->as_filter_value_list().filter_value_list();
+
+        drawing_state().filters.grow_capacity(filter_value_list.size());
+
+        // FIXME: How to get the right layout node ?
+        // auto layout_node = canvas_element().layout_node();
+
+        // 4. Set this's current filter to the given value.
+        for (auto &item : filter_value_list) {
+            item.visit(
+                [&](CSS::FilterOperation::Blur const& blur_filter) {
+                    // const float radius = blur_filter.resolved_radius(*layout_node);
```

Comment:
please remove commented code. I see that it is removed in subsequent commits but why introduce it in the first place?